core/runtime/wasm_edge/module_factory_impl.cpp

```diff
@@ -391,15 +391,28 @@ namespace kagome::runtime::wasm_edge {
   }
 
   CompilationOutcome<void> ModuleFactoryImpl::compile(
-      std::filesystem::path path_compiled, BufferView code) const {
+      std::filesystem::path path_compiled,
+      BufferView code,
+      const RuntimeContext::ContextParams &config) const {
     if (config_.exec == ExecType::Interpreted) {
       OUTCOME_TRY(writeFileTmp(path_compiled, code));
       return outcome::success();
     }
     OUTCOME_TRY(configure_ctx, configureCtx());
+    auto configure_ctx_raw = configure_ctx.raw();
+    if (configure_ctx_raw == nullptr) {
+      return CompilationError{"Configure ctx raw is nullptr"};
+    }
+
     WasmEdge_ConfigureCompilerSetOptimizationLevel(
-        configure_ctx.raw(), WasmEdge_CompilerOptimizationLevel_O3);
-    CompilerContext compiler = WasmEdge_CompilerCreate(configure_ctx.raw());
+        configure_ctx_raw, WasmEdge_CompilerOptimizationLevel_O3);
+    if (not config.wasm_ext_bulk_memory) {
+      WasmEdge_ConfigureRemoveProposal(configure_ctx_raw,
+                                       WasmEdge_Proposal_BulkMemoryOperations);
+      WasmEdge_ConfigureRemoveProposal(configure_ctx_raw,
+                                       WasmEdge_Proposal_ReferenceTypes);
+    }
```

Comment:
Duplicate (`loadCompiled`), move to `configureCtx`.

Also `WasmEdge_ConfigureRemoveProposal(WasmEdge_Proposal_ReferenceTypes)` doesn't depend on bulk proposal,
reference types proposal is disabled in polkadot-sdk.

Reply:
Disabled ReferenceTypes in cofigureCtx

Reply:
`WasmEdge_Proposal_BulkMemoryOperations` still duplicated and not moved to `configureCtx`.
Also may add `if (bulk) { WasmEdge_ConfigureAddProposal(); } else { WasmEdge_ConfigureRemoveProposal(); }` for reliability